src/core/Mayan.sol

```diff
@@ -7,65 +7,64 @@ import {SafeTransferLib} from "../vendor/SafeTransferLib.sol";
 contract Mayan {
     using SafeTransferLib for IERC20;
 
-    function bridgeERC20ToMayan(IERC20 token, address forwarder, address mayanProtocol, bytes memory protocolData) internal {
+    function bridgeERC20ToMayan(address forwarder, bytes memory protocolAndData) internal {
+        IERC20 token;
+        assembly ("memory-safe") {
+            // protocolData is abi.encode(mayanProtocol, protocolData)
+            // and protocolData is (4 bytes selector, 32 bytes token, 32 bytes amount, ...anything else)
+            // then, token is at 0x84 
```

Comment:
What's the impact of this assumption being incorrect? Do we need to check for dirty bits?